### DISCUSSION TITLE
PEP 684 Compatibility

### DISCUSSION THREAD

**Discussion by testforvln** [PyO3/pyo3#3101] [General]
Hi all,

Recently PEP 684 has been accepted, and the per-interpreter gil is aiming for 3.12 which may be released this October. Is there any plan to support this feature in PyO3?

**Comment by adamreichold** [comment]
I think we definitely would want to support it as it ties in with the general trend towards removal of global state, e.g. per-module state. But we do have significant design and implementation work to do before we achieve compatibility as we are currently using `static` variables for e.g. general caching and type objects.

**Reply by davidhewitt** [reply]
Exactly this. It's going to take a while to get there :)
